README.rst

```diff
@@ -155,3 +155,122 @@ Using your normal views over a websocket connection
+    )
+
+    class LiveConsumer(ListModelMixin, GenericAsyncAPIConsumer):
+        queryset = models.Test.objects.all()
+        serializer_class = serializers.TestSerializer
+        permission_classes = (permissions.IsAuthenticated,)
+
+
+Because this class uses the ListModelMixin, one has access to the `list` action.
+
+One can access this action from the client with a payload, or from within a method:
+
+.. code-block:: python
+
+    # From Client payload:
+    {action: "list", "request_id": 42}
+
+    # From Method:
+    data, status = await self.list()
+
+
+Note: Mixin - available action
+
+ListModelMixin - `list`
+PatchModelMixin - `patch`
+CreateModelMixin - `create`
+RetrieveModelMixin - `retrieve`
+UpdateModelMixin - `update`
+DeleteModelMixin - `delete`
+
+
+Subscribing to all instances of a model
+---------------------------------------
+
+One can subscribe to all instances of a model by utilizing the model_observer.
+
+.. code-block:: python
+
+    from djangochannelsrestframework.observer import model_observer
+
+    @model_observer(models.Test)
+    async def model_activity(self, message, observer=None, **kwargs):
+        # send activity to your frontend
+        await self.send_json(message)
+
+This method will send messages to the client on all CRUD operations made through the Django ORM.
+
+Note: These notifications do not include bulk updates, such as `models.Test.objects.filter(name="abc").update(name="newname")`
+
+
+Creating consumer operation
+---------------------------
+
+To create consumer operations, one can choose between using the traditional receive_json method utilized in typical consumers or djangochannelsrestframework actions. 
+
+Actions are created by adding the action <decorator> to a method.
+
+.. code-block:: python
+
+    from djangochannelsrestframework.decorators import action
+
+    @action()
+    async def subscribe_to_model(self, **kwargs):
+        await LiveConsumer.model_activity.subscribe(self)
+
+    async def receive_json(self, content):
+        await super().receive_json
+        await LiveConsumer.model_activity.subscribe(self)
+
+Both the action and receive_json make use of the model_activity method in the LiveConsumer class, referred to above, subscribing to all CRUD operations of the model specified in the @model_observer.
+
+Note: If utilizing `receive_json`, one must `super().receive_json` to avoid the disruption of other actions not declared in the `receive_json`.
+
+
+Initiating operation on consumer connect
+----------------------------------------
+
+One may initiate operations on consumer connects by overriding the `websocket_connect` method.
+
+.. code-block:: python
+
+    async def websocket_connect(self, message):
+        try:
+            # Super Save
+            await super().websocket_connect(message)
+
+            # Initialized operation
+            await type(self).activities_change.subscribe(self)
+
+        except Exception as e:
+            await self.close()
```

Comment:
im not sure we need to try catch if an expection is reased channels will close the connection.